auth.php

```diff
@@ -455,14 +455,15 @@ public function saml_login() {
         require('setup.php');
         require_once("$CFG->dirroot/login/lib.php");
 
-        // Set the default IdP to be the first in the list. Used when dual login is disabled.
-        $arr = array_reverse($saml2auth->metadataentities);
-        $metadataentities = array_pop($arr);
-        $idpentity = array_pop($metadataentities);
-        $idp = md5($idpentity->entityid);
+        if (isset($_GET['idp'])) {
+            $SESSION->saml2idp = $_GET['idp'];
+        }
 
-        // Specify the default IdP to use.
-        $SESSION->saml2idp = $idp;
+        // Backup in case we can't get the idp from the url param or our session idp is empty.
+        // Set the default IdP to be the first in the list. Used when dual login is disabled.
+        if (!isset($SESSION->saml2idp) || (isset($SESSION->saml2idp) && !$SESSION->saml2idp)) {
```

Comment:
This seems overly complicated, could you use !empty() ?

Reply:
Yes, I've updated that